pkg/vm/engine/tpe/tuplecodec/cubekv.go

```diff
@@ -194,35 +190,79 @@ func (ck * CubeKV) Set(key TupleKey, value TupleValue) error {
 }
 
 func (ck * CubeKV) SetBatch(keys []TupleKey, values []TupleValue) []error {
-	panic("implement me")
+	var errs []error
+	for i, key := range keys {
+		err := ck.Set(key,values[i])
+		errs = append(errs,err)
+	}
+	return errs
 }
 
 func (ck * CubeKV) DedupSet(key TupleKey, value TupleValue) error {
 	return ck.Cube.SetIfNotExist(key, value)
 }
 
 func (ck * CubeKV) DedupSetBatch(keys []TupleKey, values []TupleValue) []error {
-	panic("implement me")
+	var errs []error
+	for i, key := range keys {
+		err := ck.DedupSet(key,values[i])
+		errs = append(errs,err)
+	}
+	return errs
 }
 
 func (ck * CubeKV) Delete(key TupleKey) error {
 	return ck.Cube.Delete(key)
 }
 
+// Get gets the value of the key.
+// If the key does not exist, it returns the null
 func (ck * CubeKV) Get(key TupleKey) (TupleValue, error) {
 	return ck.Cube.Get(key)
 }
 
 func (ck * CubeKV) GetBatch(keys []TupleKey) ([]TupleValue, error) {
-	panic("implement me")
+	var values []TupleValue
```

Comment:
A little question is whether `GetBatch`/`SetBatch` is an atomic or not?

Reply:
GetBatch/SetBatch does not need atomic here. They sets/gets keys one by one.